tokamak/2-currents/BOUT.inp

```diff
@@ -0,0 +1,273 @@
+
+NOUT = 100
+TIMESTEP = 200
+
+MZ = 1     # number of points in z direction (2^n + 1)
+ZPERIOD = 1  # Number of times domain repeats in toroidal angle
+
+grid = tokamak-example.nc
+
+[restart]
+init_missing = true  # Set missing variables to 0 if missing in restarts
+
+[mesh]
+paralleltransform = shifted
+
+extrapolate_y = false # Use metrics constant into y guard cells
+
+##################################################
+# derivative methods
+
+[mesh:ddx]
+
+first = C2
+second = C2
+upwind = W3
+
+[mesh:ddy]
+
+first = C2
+second = C2
+upwind = W3
+
+[mesh:ddz]
+
+first = FFT
+second = FFT
+upwind = W3
+
+###################################################
+# Time-integration solver
+
+[solver]
+# Note: If evolving neutrals, need preconditioning
+type = cvode
+use_precon = true
+
+maxl = 10
+
+ATOL = 1.0e-10  # absolute tolerance
+RTOL = 1.0e-5   # relative tolerance
+mxstep = 1000000  # Maximum internal steps per output
+
+##################################################
+# Electrostatic potential solver
+
+[phiSolver]
+inner_boundary_flags = 0   # Zero-gradient inner boundary
+outer_boundary_flags = 16  # INVERT_SET, setting outer boundary
+
+[laplacexy]  # 2D solver in X-Y
+pctype = hypre    # Preconditioner
+
+atol = 1e-12
+rtol = 1e-8
+
+# Note: These boundary flags should be consistent with phiSolver
+core_bndry_dirichlet = false
+pf_bndry_dirichlet = true
+y_bndry_dirichlet = false
+
+include_y_derivs = true  # Include the y components in the inversion
+
+[aparSolver]
+inner_boundary_flags = 0
+outer_boundary_flags = 0
+
+all_terms = true
+nonuniform=true
+
+###################################################
+# Model settings
+
+[Hermes]
+
+#######################
+# Output variables
+output_ddt = false # Output time derivatives
+verbose = true     # Output more derived quantities
+
+#######################
+# Numerical dissipation
+
+vepsi_dissipation = true  # Parallel dissipation on Ve-Vi
+vort_dissipation = false  # Parallel dissipation of vorticity oscillations
+phi_dissipation = true    # Parallel dissipation of potential oscillations
+
+radial_buffers = true   # Turn on dissipation close to boundaries
+radial_inner_width = 4  # Number of cells on inner boundary
+radial_outer_width = -1  # Number of cell on outer boundary. < 0 -> no buffer
+radial_buffer_D = 1     # Damping coefficient in boundary
+
+#######################
+# Flux limiters
+kappa_limit_alpha = 0.2 # SOLPS style heat flux limiter
+eta_limit_alpha   = 0.5 # SOLPS style viscosity limiter
+
+#######################
+# Electric field and Ohm's law
+electromagnetic = true  # Electromagnetic? Otherwise electrostatic
+FiniteElMass    = true  # Finite electron mass?
+
+# Electrostatic potential
+split_n0 = false  # Solve n=0 electrostatic potential separately?
```

Comment:
Is this still `false` because that's more stable to get the simulations started? I'd have expected the poloidal derivatives for n=0 to be important in an axisymmetric simulation?

Reply:
It can be set to `true` but then needs PETSc which I don't have mostly. It can make quite a difference, mainly around the X-point. Perhaps a separate input with these options turned on would be good.